patroni/validator.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+import logging
+import os
+import socket
+
+from patroni.utils import split_host_port
+
+logger = logging.getLogger(__name__)
+
+
+def data_directory_empty(data_dir):
+    if os.path.isfile(os.path.join(data_dir, "global", "pg_control")):
+        return False
+    if not os.path.exists(data_dir):
+        return True
+    return all(
+        os.name != "nt" and (n.startswith(".") or n == "lost+found")
+        for n in os.listdir(data_dir)
+    )
```

Comment:
It pretty much duplicates `Postgresql.data_directory_empty()`

Reply:
yeah)

Reply:
Sure, as soon as I come up with a solution.